web/libs/core/src/blocks/TokenSettingsModal/index.tsx

```diff
@@ -13,18 +16,27 @@ export const TokenSettingsModal = ({
 }) => {
   const settings = useAtomValue(settingsAtom);
   const formRef = useRef<Form>();
-  const { mutate: saveSettings } = useAtomValue(saveSettingsAtom);
-  const onSubmit = debounce(() => {
+  const { mutate: saveSettings, isPending, isSuccess, isIdle } = useAtomValue(saveSettingsAtom);
+  const [saved, setSaved] = useState(false);
+  const onSubmit = () => {
     const form = formRef.current.formElement.current as HTMLFormElement;
     const data = formDataToJPO(new FormData(form));
     const body = {
+      ...data,
       api_tokens_enabled: data.api_tokens_enabled === "on",
       legacy_api_tokens_enabled: data.legacy_api_tokens_enabled === "on",
     } satisfies Partial<AuthTokenSettings>;
     saveSettings(body);
-  });
+    setSaved(true);
+  };
+
+  useEffect(() => {
+    if (saved) {
+      setTimeout(() => setSaved(false), 2000);
```

Comment:
and `Form.Indicator` instead of this?

Reply:
if we're going down the timeout approach we should also clear time out in cases where we set the time out again - although given the context is not a big deal - using Form.Indicator is likely the better option as we should double down on existing code